test/has_shebang.bats

```diff
@@ -195,6 +195,22 @@ setup () {
   done
 }
 
+@test "has_shebang() - SHELL DIRECTIVE" {
+  source "${PROJECT_ROOT}/src/functions.sh"
+
+  local interpreters=(
+    {,a,ba,da,k}sh
+    'bats'
+  )
+
+  for i in "${interpreters[@]}"; do
+    echo -e "#!/bin/$i\n# shellcheck shell=$i\n\nshell" > script
```

Comment:
Thank you, one small suggestion. I would like to see something like this, so we know that if that checks shellcheck directives that decides if it's a shell script or not.

```suggestion
    echo -e "#!/bin/unknown\n# shellcheck shell=$i\n\nshell" > script
```

Reply:
Barely beat you to it https://github.com/redhat-plumbers-in-action/differential-shellcheck/pull/104/commits/ce6112490968623e53cc30192d3775190979ea71

Reply:
resolved in ce61124